src/functions/auth.js

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios'
+
+export const handler = async (event, _, callback) => {
+  try {
+    _retrieveToken(event, callback)
+  } catch (e) {
+    callback(Error(e), {
+      status: 500,
+      body: 'Server Error',
+    })
+  }
+}
+
+function _retrieveToken(event, callback) {
+  const { code, state, redirect_uri } = event.queryStringParameters
+  if (!code || !state || !redirect_uri) {
+    throw new Error('Missing parameters')
```

Comment:
Don't actually think we need `redirect_uri` at this step. Only for the client-side call to the `authorize` endpoint.